### PR TITLE
[feat] pull and display all plants

### PR DESCRIPTION
**Pull request by kylezryr**
### Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."
- Created Supabase client 
- Created Plant type based on schema
- Created getPlantSeasonality query on Supabase and on TS

### Screenshots
[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy! Use the provided image template. Drag the desired image into the PR, then copy the link into the placeholder."

[image placeholder]: 

## How to review
[//]: # 'Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'

## Next steps
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."

### Online sources
[//]: # 'Copy links to any tutorials or documentation that was useful to you when working on this PR'

### Related PRs
[//]: # "Add related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

CC: @ccatherinetan

### CODE REVIEW

File: components/PlantList.tsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import React, { useEffect, useState } from 'react';
```
Comment (ccatherinetan) [inline components/PlantList.tsx:1]: this looks great! sorry for the change, but is it possible to create the `seasonal-planting-guide` page (route: `app/seasonal-planting-guide`) that uses the `PlantList` component?

**Review by ccatherinetan — COMMENTED** [review]
looking great kyle! 
1. my main request is to create the `seasonal-planting-guide` page to test out the `PlantList` component. 
2. lowk, we might change the schema a bit, so we might have to update that today during worksesh too. more soon!
3. have you alr rebased? i'm not sure if there's a way to not add the createClient file again since it alr exists? 
4. let's rename this pr to "[feat] pull and display all plants"
